client/rpc_client.go

```diff
@@ -100,6 +103,9 @@ func (c *JSONRPCClient) Call(method string, params map[string]interface{}, id st
 	if err != nil {
 		return err
 	}
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("post status not OK: %s", resp.Status)
```

Comment:
You need to close the body even if it’s a failure and likely you want to return the body in the error message